src/Mechanisms/RenderComponent.php

```diff
@@ -14,26 +14,17 @@ function boot()
 
     public static function livewire($expression)
     {
-        $key = "'" . Str::random(7) . "'";
+        $key = app(\Livewire\Mechanisms\ExtendBlade\DeterministicBladeKeys::class)->generate();
 
-        $pattern = "/,\s*?key\(([\s\S]*)\)/"; //everything between ",key(" and ")"
+        $key = "'{$key}'";
+
+        $pattern = "/,\s*?key\(([\s\S]*)\)/"; // everything between ",key(" and ")"
 
         $expression = preg_replace_callback($pattern, function ($match) use (&$key) {
             $key = trim($match[1]) ?: $key;
             return "";
         }, $expression);
```

Comment:
```suggestion
        $key = null;
        $pattern = "/,\s*?key\(([\s\S]*)\)/"; // everything between ",key(" and ")"

        $expression = preg_replace_callback($pattern, function ($match) use (&$key) {
            $key = trim($match[1]);
            return "";
        }, $expression);
        
        if (! $key) {
                $key = app(\Livewire\Mechanisms\ExtendBlade\DeterministicBladeKeys::class)->generate();
                $key = "'{$key}'";
        }
```

maybe something like this